File: backend/app/schemas/place.py
"""Place API schemas. PlaceSummary is the lean map-pin/card shape; PlaceDetail adds
the full Google base data shown when a pin is opened.
"""
from __future__ import annotations

import uuid

from pydantic import BaseModel, ConfigDict


class PlaceSummary(BaseModel):
    """The shared Google base data shown on a recommendation card / map pin."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    google_place_id: str
    name: str
    address: str | None = None
    lat: float | None = None
    lng: float | None = None
    rating: float | None = None
    user_rating_count: int | None = None
    google_maps_uri: str | None = None


class PlaceDetailResponse(PlaceSummary):
    """Full shared place detail (everything Google gives us for the pin sheet)."""

    phone: str | None = None
    opening_hours: dict | None = None
    photos: list | None = None

File: packages/types/src/place.ts
// Mirrors backend app/schemas/place.py and the Google match candidate.

export interface PlaceSummary {
  id: string;
  google_place_id: string;
  name: string;
  address: string | null;
  lat: number | null;
  lng: number | null;
  rating: number | null;
  user_rating_count: number | null;
  google_maps_uri: string | null;
}

export interface PlaceDetail extends PlaceSummary {
  phone: string | null;
  opening_hours: Record<string, unknown> | null;
  photos: unknown[] | null;
}

// Result of GET /places/search (Google Places proxy).
export interface PlaceCandidate {
  google_place_id: string;
  name: string;
  address: string | null;
  lat: number | null;
  lng: number | null;
  rating: number | null;
  user_rating_count: number | null;
  google_maps_uri: string | null;
}
